[FLINK-14907]  [filesystem] Make Azure Storage Key configurable via Environment Variable

knaufk opened the pull request:
## What is the purpose of the change

Make Azure Storage Key configurable via Environment Variable


## Brief change log
* update documentation on credential configuration for ABS
* add EnvironmentVariableKeyProvider for Azure Blob Storage

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - build Flink
  - set Flink configuration as outlined in documentation
  - set AZURE_STORAGE_KEY environment variable
  - use wasbs:// for checkpoints or as file input for a testing job
  - (test_azure_fs.sh can be modified to test this)

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? docs, JavaDocs

StephanEwen commented:
Looks good, +1 to merge.

Nit: I think the optional wrapping here is a bit overkill. For such a local variable, just checking for null seems even easier.

knaufk commented:
@StephanEwen Thank you for the review. I have addressed your comment and fixed some typos in the documentation.

It would help, if we can still merge this before the release branch is cut.

StephanEwen commented:
Merging this...
